apis/services/cartService.js

```diff
@@ -0,0 +1,54 @@
+const { cartDao } = require('../models');
+const { checkIfTheCartExists } = require('../models/cartDao');
+const { checkStock } = require('../utils/checkStock');
+
+const getCartsByUserId = async (userId) => {
+  const cart = await cartDao.getCartsByUserId(userId);
+
+  return await cart;
+};
+
+const postCart = async (productOptionId, quantity, userId) => {
+  const checkIfTheCartExists = await cartDao.checkIfTheCartExists(
+    productOptionId,
+    userId
+  );
+  if (checkIfTheCartExists === undefined) {
+    await checkStock(productOptionId, quantity);
+
+    return await cartDao.postCart(productOptionId, userId, quantity);
+  } else if (
+    checkIfTheCartExists.productOptionId === parseInt(productOptionId)
+  ) {
+    await checkStock(productOptionId, quantity);
+
+    return await cartDao.updateQuantityWhenPostCart(
```

Comment:
위의 if 조건을 만족 할 경우 return에 의해 함수가 종료 되므로 아래에 else if 보다는 if를 사용하는 것이 조금 더 가독성에 좋습니다.
```suggestion
  }
  
  if (checkIfTheCartExists.productOptionId === parseInt(productOptionId) {
    await checkStock(productOptionId, quantity);

    return await cartDao.updateQuantityWhenPostCart(
```

Reply:
완료